### PR TITLE
feat(patterns/icons): Add color icons

### PR DESCRIPTION
**Pull request by gael-boyenval**
## I have read [the contributing guidelines](https://mozaic.adeo.cloud/Contributing/)

- [x] Yes
- [ ] No

## Does this PR introduce a breaking change?

read [What is a breaking change ?](https://mozaic.adeo.cloud/Contributing/Prerequisite/GitConventions/#breaking-changes-)

- [ ] Yes
- [x] No

## Describe the changes

Github issue number or Jira issue URL: https://design-system-adeo.atlassian.net/browse/MZC-142

## Other informations

Please review commit after commit since there is a lot of elements.
The PR surely lacks documentation.

### CODE REVIEW

File: iconCompiler/config.js [hunk 1]
```diff
@@ -1,10 +1,22 @@
...
-    svg: path.join(currentDir, 'packages/icons/svg/'),
-    react: path.join(currentDir, 'packages/icons/react/'),
-    vue: path.join(currentDir, 'packages/icons/vue/'),
+  sourcePaths: {
+    svg: destPath('src/icons/SVG'),
+    svgColor: destPath('src/icons/COLORS/SVG'),
+    svgColorToGatsby: destPath('src/icons/COLORS/SVG'),
+    pdf: destPath('src/icons/PDF'),
+    pdfColor: destPath('src/icons/COLORS/PDF'),
+  },
```
Comment (mohamedMok) [inline iconCompiler/config.js]: Any reason to use full caps for folder names?

File: iconCompiler/copyPDF/pdfValidator.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      )
+    )
+  }
+
+  if (!pdfName.includes(`${sizeDirectory}.pdf`)) {
+    errors.push(
+      error(
+        pdfName,
+        ` found in /${sizeDirectory}/ directory. Icon File name is either wrong or the icon is in the wrong directory`
+      )
```
Comment (mohamedMok) [inline iconCompiler/copyPDF/pdfValidator.js]: Need some consistency on error. 
2 # way for `Icon filename`
`icon Filename`
`Icon File name`
both are correct!!

File: iconCompiler/config.js (hunk 1 above)
Comment (gael-boyenval) [reply iconCompiler/config.js]: it's the directory structure given by the designers.

File: src/docs/Foundations/Icons/catalog.mdx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ---
-title: 'Preview'
-order: 3
+title: 'Monochroms icons'
```
Comment (caqueste) [inline src/docs/Foundations/Icons/catalog.mdx]: ~Monochroms icons~ Monochrom icons

File: src/docs/Foundations/Icons/designers.mdx [hunk 4]
```diff
@@ -1,11 +1,12 @@
 ---
 title: 'Design'
-order: 4
+order: 5
 ---
 
->Icons must be **recognisable at first sight**. Here you will find how Leroy Merlin's icon kit has been made along with some design specificities to help you understand our vision.
+> Icons must be **recognisable at first sight**. Here you will find how Leroy Merlin's icon kit has been made along with some design specificities to help you understand our vision.
```
Comment (caqueste) [inline src/docs/Foundations/Icons/designers.mdx]: ~Leroy Merlin~ LEROY MERLIN

File: src/docs/Foundations/Icons/designers.mdx [hunk 5]
```diff
@@ -33,6 +37,7 @@ We have defined 5 sizes for our icons :
 In some very particular cases a 48px can be resized to 40px. 16px icons can only be used for navigation and rating stars.
 
 ### 2. Thickness
+
 Leroy Merlin's icon are **outlined**. Thickness is very important in this style and must be followed. Every icon size has its guidelines in terms of thickness. Icon's thickness are divided in two categories : **major thickness** used to draw the outer shape of the icon and **minor thickness** to add a second level of details in the icon.
```
Comment (caqueste) [inline src/docs/Foundations/Icons/designers.mdx]: ~Leroy Merlin~ LEROY MERLIN

**Comment by gael-boyenval** [comment]
@caqueste : added documentation on icon contributing rules for designer

File: iconCompiler/config.js (hunk 1 above)
Comment (mohamedMok) [reply iconCompiler/config.js]: Copy that 👍

**Pull request merged**